Statement in the database error

_What version of AdvancedBan (`/AdvancedBan`) are you using?_

 §8[]=====[§7Enabling AdvancedBan§8]=====[]
§8| §cInformation:
§8|   §cName: §7AdvancedBan
§8|   §cDeveloper: §7Leoko
§8|   §cVersion: §72.1.5-SNAPSHOT
§8|   §cStorage: §7MySQL (external)
§8| §cSupport:
§8|   §cSkype: §7Leoko33
§8|   §cMail: §7Leoko4433@gmail.com
§8|   §cGithub: §7https://github.com/DevLeoko/AdvancedBan/issues
§8|   §cDiscord: §7https://discord.gg/ycDG6rS
§8| §cUpdate:
§8|   §7There is a new version available! [2.1.4]
§8[]================================[]§f
 
_What kind of server do you have (`Bungeecord/single server`)?_
 Bungeecord
 
_What server version (/version) are you using?_

??9This server is running BungeeCord version git:BungeeCord-Bootstrap:1.12-SNAPSHOT:fde2c3f:1310 by md_5

_Please povide the EXACT steps required to reproduce the problem..._
1.
2.
3.

_Any error/log post it through [pastebin.com](http://pastebin.com) and link it here._

https://pastebin.com/QQNkj8MF

_Add any additional information below._

Latest.log:
https://pastebin.com/k8qvn0XR

This error is still around...

```
>AdvancedBan
[17:10:33 INFO]: -===== AdvancedBan v2 =====-
[17:10:33 INFO]:   Dev ? Leoko
[17:10:33 INFO]:   Status ? Stable
[17:10:33 INFO]:   Version ? 2.1.5-RELEASE
[17:10:33 INFO]:   License ? Public
[17:10:33 INFO]:   Storage ? MySQL (external)
[17:10:33 INFO]:   Server ? Spigot/Bukkit
[17:10:33 INFO]:   UUID-Mode ? INTERN
[17:10:33 INFO]:   Prefix ? AdvancedBan ?
[17:10:33 INFO]: -=========================-
>version
[17:11:08 INFO]: This server is running CraftBukkit version git-Spigot-642f6d2-57ab4cf (MC: 1.12.2) (Implementing API version 1.12.2-R0.1-SNAPSHOT)
```

```
[16:22:39 INFO]: [AdvancedBan] An unexpected error has ocurred executing an Statement in the database
Please check the plugins/AdvancedBan/logs/latest.log file and report thiserror in: https://github.com/DevLeoko/AdvancedBan/issues
[16:22:39 ERROR]: Could not pass event AsyncPlayerPreLoginEvent to AdvancedBan v2.1.5-RELEASE
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) [spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:482) [spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        at net.minecraft.server.v1_12_R1.LoginListener$LoginHandler.fireEvents(LoginListener.java:271) [spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        at net.minecraft.server.v1_12_R1.LoginListener$3.run(LoginListener.java:224) [spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
Caused by: java.lang.NullPointerException
        at me.leoko.advancedban.manager.PunishmentManager.load(PunishmentManager.java:47) ~[?:?]
        at me.leoko.advancedban.Universal.callConnection(Universal.java:212) ~[?:?]
        at me.leoko.advancedban.bukkit.listener.ConnectionListener.onConnect(ConnectionListener.java:20) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_171]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_171]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_171]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_171]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        ... 5 more
```

```
[16:22:39] An unexpected error has ocurred executing an Statement in the database
Please check the plugins/AdvancedBan/logs/latest.log file and report thiserror in: https://github.com/DevLeoko/AdvancedBan/issues
[16:22:39] Query: 
SELECT * FROM `Punishments` WHERE `uuid` = ? OR `uuid` = ?
```

--- a/src/main/java/me/leoko/advancedban/manager/DatabaseManager.java
+++ b/src/main/java/me/leoko/advancedban/manager/DatabaseManager.java
@@ -18,6 +18,8 @@ import java.sql.*;
  */
 public class DatabaseManager {
 
+	private final Object lock = new Object();
+	
     private String ip;
     private String dbName;
     private String usrName;
@@ -162,35 +164,37 @@ public class DatabaseManager {
     }
 
     private ResultSet executeStatement(String sql, boolean result, Object... parameters) {
-        try (PreparedStatement statement = connection.prepareStatement(sql)) {
-
-            for (int i = 0; i < parameters.length; i++) {
-                Object obj = parameters[i];
-                if (obj instanceof Integer) {
-                    statement.setInt(i + 1, (Integer) obj);
-                } else if (obj instanceof String) {
-                    statement.setString(i + 1, (String) obj);
-                } else if (obj instanceof Long) {
-                    statement.setLong(i + 1, (Long) obj);
-                } else {
-                    statement.setObject(i + 1, obj);
-                }
-            }
-
-            if (result) {
-                return statement.executeQuery();
-            }
-			statement.execute();
-        } catch (SQLException ex) {
-            Universal.get().log(
-                    "An unexpected error has occurred executing an Statement in the database\n"
-                    + "Please check the plugins/AdvancedBan/logs/latest.log file and report this"
-                    + "error in: https://github.com/DevLeoko/AdvancedBan/issues"
-            );
-            Universal.get().debug("Query: \n" + sql);
-            Universal.get().debug(ex);
-        }
-        return null;
+    	synchronized (lock) {
+    		try (PreparedStatement statement = connection.prepareStatement(sql)) {
+    			
+    			for (int i = 0; i < parameters.length; i++) {
+    				Object obj = parameters[i];
+    				if (obj instanceof Integer) {
+    					statement.setInt(i + 1, (Integer) obj);
+    				} else if (obj instanceof String) {
+    					statement.setString(i + 1, (String) obj);
+    				} else if (obj instanceof Long) {
+    					statement.setLong(i + 1, (Long) obj);
+    				} else {
+    					statement.setObject(i + 1, obj);
+    				}
+    			}
+    			
+    			if (result) {
+    				return statement.executeQuery();
+    			}
+    			statement.execute();
+    		} catch (SQLException ex) {
+    			Universal.get().log(
+    					"An unexpected error has occurred executing an Statement in the database\n"
+    							+ "Please check the plugins/AdvancedBan/logs/latest.log file and report this"
+    							+ "error in: https://github.com/DevLeoko/AdvancedBan/issues"
+    					);
+    			Universal.get().debug("Query: \n" + sql);
+    			Universal.get().debug(ex);
+        	}
+        	return null;
+    	}
     }
 
     /**

--- a/src/main/java/me/leoko/advancedban/manager/DatabaseManager.java
+++ b/src/main/java/me/leoko/advancedban/manager/DatabaseManager.java
@@ -168,16 +168,7 @@ public class DatabaseManager {
     		try (PreparedStatement statement = connection.prepareStatement(sql)) {
     			
     			for (int i = 0; i < parameters.length; i++) {
-    				Object obj = parameters[i];
-    				if (obj instanceof Integer) {
-    					statement.setInt(i + 1, (Integer) obj);
-    				} else if (obj instanceof String) {
-    					statement.setString(i + 1, (String) obj);
-    				} else if (obj instanceof Long) {
-    					statement.setLong(i + 1, (Long) obj);
-    				} else {
-    					statement.setObject(i + 1, obj);
-    				}
+    				statement.setObject(i + 1, parameters[i]);
     			}
     			
     			if (result) {

--- a/src/main/java/me/leoko/advancedban/manager/DatabaseManager.java
+++ b/src/main/java/me/leoko/advancedban/manager/DatabaseManager.java
@@ -17,8 +17,6 @@ import java.sql.*;
  * {@link PunishmentManager#getPunishmentFromResultSet(ResultSet)} for already parsed data.
  */
 public class DatabaseManager {
-
-	private final Object lock = new Object();
 	
     private String ip;
     private String dbName;
@@ -163,29 +161,27 @@ public class DatabaseManager {
         return executeStatement(sql.toString(), result, parameters);
     }
 
-    private ResultSet executeStatement(String sql, boolean result, Object... parameters) {
-    	synchronized (lock) {
-    		try (PreparedStatement statement = connection.prepareStatement(sql)) {
-    			
-    			for (int i = 0; i < parameters.length; i++) {
-    				statement.setObject(i + 1, parameters[i]);
-    			}
-    			
-    			if (result) {
-    				return statement.executeQuery();
-    			}
-    			statement.execute();
-    		} catch (SQLException ex) {
-    			Universal.get().log(
-    					"An unexpected error has occurred executing an Statement in the database\n"
-    							+ "Please check the plugins/AdvancedBan/logs/latest.log file and report this"
-    							+ "error in: https://github.com/DevLeoko/AdvancedBan/issues"
-    					);
-    			Universal.get().debug("Query: \n" + sql);
-    			Universal.get().debug(ex);
-        	}
-        	return null;
-    	}
+    private synchronized ResultSet executeStatement(String sql, boolean result, Object... parameters) {
+    	try (PreparedStatement statement = connection.prepareStatement(sql)) {
+    		
+    		for (int i = 0; i < parameters.length; i++) {
+    			statement.setObject(i + 1, parameters[i]);
+    		}
+   			
+    		if (result) {
+    			return statement.executeQuery();
+    		}
+   			statement.execute();
+    	} catch (SQLException ex) {
+    		Universal.get().log(
+   					"An unexpected error has occurred executing an Statement in the database\n"
+   							+ "Please check the plugins/AdvancedBan/logs/latest.log file and report this"
+    						+ "error in: https://github.com/DevLeoko/AdvancedBan/issues"
+    				);
+    		Universal.get().debug("Query: \n" + sql);
+    		Universal.get().debug(ex);
+       	}
+        return null;
     }
 
     /**

